fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/ScalarValuedFunction.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * UDF.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2025 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.metadata;
+
+import com.apple.foundationdb.record.PlanHashable;
+import com.apple.foundationdb.record.PlanSerializationContext;
+import com.apple.foundationdb.record.RecordMetaDataProto;
+import com.apple.foundationdb.record.query.plan.cascades.MacroFunction;
+import com.apple.foundationdb.record.query.plan.serialization.DefaultPlanSerializationRegistry;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Defines a scalar User-defined-function.
+ */
+public class ScalarValuedFunction {
```

Comment:
Why don't we just make a trait-like interface called `SerializableFunction` that makes this proto-serializable? `MacroFunction` can then implement `SerializableFunction` Then we don't have to wrap the `MacroFunction` here. It would also allow us to
support other functions in the future that are not `MacroFunction`. It's pretty straightforward for `toProto(...)`, slightly less so on the way back. Please look how the plan serialization achieves the same thing by using:
1. `oneof` in the proto declarations
2. `PlanDeserializer` using `AutoService` to get the  dispatch of the from proto part right